router/batchrouter/handle.go

```diff
@@ -0,0 +1,841 @@
+			_ = gzWriter.WriteGZ(line)
+		}
+	}
+	_ = gzWriter.CloseGZ()
+	if !eventsFound {
+		brt.logger.Infof("BRT: No events in this batch for upload to %s. Events are either de-deuplicated or skipped", provider)
+		return UploadResult{
+			LocalFilePaths: []string{gzipFilePath},
+		}
+	}
+	// assumes events from warehouse have receivedAt in metadata
+	var firstEventAt, lastEventAt string
+	if isWarehouse {
+		firstEventAtStr := gjson.GetBytes(batchJobs.Jobs[0].EventPayload, "metadata.receivedAt").String()
+		lastEventAtStr := gjson.GetBytes(batchJobs.Jobs[len(batchJobs.Jobs)-1].EventPayload, "metadata.receivedAt").String()
+
+		// received_at set in rudder-server has timezone component
+		// whereas first_event_at column in wh_staging_files is of type 'timestamp without time zone'
+		// convert it to UTC before saving to wh_staging_files
+		firstEventAtWithTimeZone, err := time.Parse(misc.RFC3339Milli, firstEventAtStr)
+		if err != nil {
+			brt.logger.Errorf(`BRT: Unable to parse receivedAt in RFC3339Milli format from eventPayload: %v. Error: %v`, firstEventAtStr, err)
+		}
+		lastEventAtWithTimeZone, err := time.Parse(misc.RFC3339Milli, lastEventAtStr)
+		if err != nil {
+			brt.logger.Errorf(`BRT: Unable to parse receivedAt in RFC3339Milli format from eventPayload: %v. Error: %v`, lastEventAtStr, err)
+		}
+
+		firstEventAt = firstEventAtWithTimeZone.UTC().Format(time.RFC3339)
+		lastEventAt = lastEventAtWithTimeZone.UTC().Format(time.RFC3339)
+	} else {
+		firstEventAt = gjson.GetBytes(batchJobs.Jobs[0].EventPayload, "receivedAt").String()
+		lastEventAt = gjson.GetBytes(batchJobs.Jobs[len(batchJobs.Jobs)-1].EventPayload, "receivedAt").String()
+	}
+
+	brt.logger.Debugf("BRT: Logged to local file: %v", gzipFilePath)
+	useRudderStorage := isWarehouse && misc.IsConfiguredToUseRudderObjectStorage(batchJobs.Connection.Destination.Config)
+	uploader, err := brt.fileManagerFactory.New(&filemanager.SettingsT{
+		Provider: provider,
+		Config: misc.GetObjectStorageConfig(misc.ObjectStorageOptsT{
+			Provider:         provider,
+			Config:           batchJobs.Connection.Destination.Config,
+			UseRudderStorage: useRudderStorage,
+			WorkspaceID:      batchJobs.Connection.Destination.WorkspaceID,
+		}),
+	})
+	if err != nil {
+		return UploadResult{
+			Error:          err,
+			LocalFilePaths: []string{gzipFilePath},
+		}
+	}
+
+	outputFile, err := os.Open(gzipFilePath)
+	if err != nil {
+		panic(err)
+	}
+
+	brt.logger.Debugf("BRT: Starting upload to %s", provider)
+	folderName := ""
+	if isWarehouse {
+		folderName = config.GetString("WAREHOUSE_STAGING_BUCKET_FOLDER_NAME", "rudder-warehouse-staging-logs")
+	} else {
+		folderName = config.GetString("DESTINATION_BUCKET_FOLDER_NAME", "rudder-logs")
+	}
+
+	var datePrefixLayout string
+	if brt.datePrefixOverride != "" {
+		datePrefixLayout = brt.datePrefixOverride
+	} else {
+		dateFormat, _ := brt.dateFormatProvider.GetFormat(brt.logger, uploader, batchJobs.Connection, folderName)
```

Comment:
@achettyiitr Do we still need this part of the code? I remember this was something temporary done a while back